compiler/pkg/generator/graphql_common.go

```diff
@@ -45,18 +47,33 @@ func jsonMarshalResolver(FieldName, PkgName string) string {
 	return fmt.Sprintf("%s, _ := json.Marshal(v%s.Spec.%s)\n%sData := string(%s)\n", FieldName, PkgName, FieldName, FieldName, FieldName)
 }
 
-func validateImportPkg(pkgName, typeString string, importMap map[string]string) (string, string) {
+func getPkgName(pkgs parser.Packages, pkgPath string) string {
+	importPath, err := strconv.Unquote(pkgPath)
+	if err != nil {
+		log.Errorf("Failed to parse the package path : %s: %v", pkgPath, err)
+	}
+	return pkgs[importPath].Name
+}
+
+func ValidateImportPkg(pkgName, typeString string, importMap map[string]string, pkgs parser.Packages) (string, string) {
 	typeWithoutPointers := strings.ReplaceAll(typeString, "*", "")
 	if strings.Contains(typeWithoutPointers, ".") {
 		part := strings.Split(typeWithoutPointers, ".")
 		if val, ok := importMap[part[0]]; ok {
-			pkgName := val[strings.LastIndex(val, "/")+1 : len(val)-1]
+			pkgName := getPkgName(pkgs, val)
 			repName := strings.ReplaceAll(pkgName, "-", "")
-			return repName + "_" + part[1], cases.Title(language.Und, cases.NoLower).String(repName) + cases.Title(language.Und, cases.NoLower).String(part[1])
+			return fmt.Sprintf("%s_%s", repName, part[1]), cases.Title(language.Und, cases.NoLower).String(repName) + cases.Title(language.Und, cases.NoLower).String(part[1])
```

Comment:
can you create single function for `fmt.Sprintf("%s_%s", strings.ToLower(pkgName), part[1]), cases.Title(language.Und, cases.NoLower).String(pkgName) + cases.Title(language.Und, cases.NoLower).String(part[1])`

Reply:
yes addressed